source/lambda/kendraIndexPoller/lambda_function.py

```diff
@@ -0,0 +1,34 @@
+import boto3
+
+def on_create(event, context):
+    kendra_client = boto3.client('kendra')
+    # get status of index
+    response = kendra_client.describe_index(Id=event['PhysicalResourceId'])
+    is_complete = False
+    status = response['Status']
+    if status == "CREATING":
+        print("still creating index")
+    elif status == "ACTIVE":
+        print("Kendra index is now active")
+        return { 'IsComplete': True }
+    elif status == "FAILED":
+        print("Kendra index creation has failed")
```

Comment:
If it is failed , shouldn't return an error case?

Reply:
If the creation failed, the operation is still considered complete, hence I returned true in that case as well. We should probably ask the customer to check logs for errors in this case. I will update that.

Reply:
I dont think we can ask the customer to look at the logs. 
We should be able to throw an error and show the reason within CloudFormation itself using a "Reason" attribute
See the FAILED section in this [link](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/crpg-ref-requesttypes-create.html)